network/test/middleware_test.go

```diff
@@ -805,6 +811,80 @@ func (m *MiddlewareTestSuite) TestUnsubscribe() {
 	unittest.RequireNeverReturnBefore(m.T(), msgRcvdFun, 2*time.Second, "message received unexpectedly")
 }
 
+// TestUnicastStreamLimit tests that the middleware does not accept more than the maximum number of
+// unicast streams and streams created beyond the limit are reset
+func (m *MiddlewareTestSuite) TestUnicastStreamLimit() {
```

Comment:
Following my previous comment, would you please also add a concurrent version of this test? i.e., creating streams concurrently and also closing them concurrently.

Reply:
b5c22ed